xla/service/gpu/runtime3/command_buffer_cmd.h

```diff
@@ -163,6 +164,27 @@ class MemcpyDeviceToDeviceCmd : public CommandBufferCmd {
   int64_t num_bytes_;
 };
 
+//===----------------------------------------------------------------------===//
+// AllocateCmd
+//===----------------------------------------------------------------------===//
+
+class AllocateCmd : public CommandBufferCmd {
+ public:
+  AllocateCmd(int64_t size, BufferAllocation::Slice dst, se::DeviceMemoryBase* dptr);
```

Comment:
I'm not sure `Slice` make sense here, this looks like the correct API;

```
AllocateCmd(int64_t size, BufferAllocation::Index buffer_index);
```

```
Record(... params, ...) {

se::DeviceMemoryBase& memory = params->allocations->GetMutableDeviceAddress(buffer_index);
... create allocation node
memory = se::DeviceMemoryBase(allocated_ptr, size);
}
```

With this API some of the buffer allocations in XLA executable will become "lazily allocated"